tests/framework/artifacts.py

```diff
@@ -1,220 +1,87 @@
 # Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 # SPDX-License-Identifier: Apache-2.0
-"""Define classes for interacting with CI artifacts in s3."""
 
-import functools
-import os
-import platform
-import tempfile
+"""Define classes for interacting with CI artifacts"""
+
+import re
 from dataclasses import dataclass
-from enum import Enum
 from pathlib import Path
-from shutil import copyfile
-from stat import S_IREAD, S_IWRITE
+from typing import Iterator
 
-import boto3
-import botocore.client
+import pytest
 
 import host_tools.network as net_tools
-from framework.defs import (
-    DEFAULT_TEST_SESSION_ROOT_PATH,
-    SUPPORTED_KERNELS,
-    SUPPORTED_KERNELS_NO_SVE,
-)
-from framework.utils import compare_versions, get_kernel_version
-from framework.utils_cpuid import get_instance_type
-from host_tools.cargo_build import run_rebase_snap_bin
-
-ARTIFACTS_LOCAL_ROOT = f"{DEFAULT_TEST_SESSION_ROOT_PATH}/ci-artifacts"
+from framework.defs import ARTIFACT_DIR
+from framework.properties import global_props
+from framework.utils import get_firecracker_version_from_toml
 
 
 def select_supported_kernels():
-    """Select guest kernels supported by the current combination of kernel and instance type."""
-    supported_kernels = SUPPORTED_KERNELS
-    host_kernel_version = get_kernel_version(level=1)
-    try:
-        instance_type = get_instance_type()
-    # in case we are not in EC2, return the default
-    # pylint: disable=broad-except
-    except Exception:
-        return supported_kernels
-
-    if instance_type == "c7g.metal" and host_kernel_version == "4.14":
-        supported_kernels = SUPPORTED_KERNELS_NO_SVE
-
+    """Select guest kernels supported by the current combination of kernel and
+    instance type.
+    """
+    supported_kernels = [r"vmlinux-4.14.\d+"]
+    if (
+        global_props.instance == "c7g.metal"
+        and global_props.host_linux_version == "4.14"
+    ):
+        supported_kernels.append(r"vmlinux-5.10-no-sve.bin")
+    else:
+        supported_kernels.append(r"vmlinux-5.10.\d+")
     return supported_kernels
 
 
-class ArtifactType(Enum):
-    """Supported artifact types."""
+def kernels(glob) -> Iterator:
+    """Return supported kernels as kernels supported by the current combination of kernel and
+    instance type.
```

Comment:
I think the "supported kernels as" part is superfluous? Otherwise I'm not sure what this comment wants to tell me :(